### PR TITLE
Add OpenBLAS corename detection

### PR DESCRIPTION
**Pull request by ogrisel**
Demo:

```json
$ python -m threadpoolctl -i numpy
[
  {
    "filepath": "/Users/ogrisel/miniforge3/envs/dev/lib/libopenblasp-r0.3.12.dylib",
    "prefix": "libopenblas",
    "user_api": "blas",
    "internal_api": "openblas",
    "version": "0.3.12",
    "num_threads": 8,
    "threading_layer": "openmp",
    "corename": "armv8"
  },
  {
    "filepath": "/Users/ogrisel/miniforge3/envs/dev/lib/libomp.dylib",
    "prefix": "libomp",
    "user_api": "openmp",
    "internal_api": "openmp",
    "version": null,
    "num_threads": 8
  }
]
```

### CODE REVIEW

**Comment by jeremiedbb** [comment]
worth to add it for BLIS as well ? https://github.com/flame/blis/blob/master/docs/BLISTypedAPI.md#specific-configuration
I didn't find an equivalent for mkl.

**Comment by ogrisel** [comment]
Shall we rename the attribute to `architecture` instead of `corename` then?

**Comment by jeremiedbb** [comment]
you have a segfault :/

File: tests/test_threadpoolctl.py [hunk 1]
```diff
@@ -440,3 +440,22 @@ def test_command_line_import_flag():
         assert "WARNING: could not import scipy.linalg" in warnings
     else:
         assert "WARNING: could not import scipy.linalg" not in warnings
+
+
+def test_architecture():
+    expected_openblas_architectures = (
+        # XXX: add more as needed by CI or developer laptops
+        "armv8",
+        "haswell",
```
Comment (jeremiedbb) [inline tests/test_threadpoolctl.py:449]:
```suggestion
        "Haswell",
```

File: tests/test_threadpoolctl.py [hunk 2]
```diff
@@ -440,3 +440,22 @@ def test_command_line_import_flag():
...
+        "SkylakeX",
+        "Sandybridge",
+    )
+    expected_blis_architectures = (
+        # XXX: add more as needed by CI or developer laptops
+        "skx",
```
Comment (jeremiedbb) [inline tests/test_threadpoolctl.py:455]:
```suggestion
        "skx",
        "haswell",
```

**Review by jeremiedbb — APPROVED** [review]

**Pull request merged**